src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,410 @@
+                Channel.create(
+                    name=elements[idx],
+                    excitation_wavelength=elements[idx + 1],
+                    excitation_power=elements[idx + 2],
+                    exposure_time=elements[idx + 3],
+                    additional_focus_offset=elements[idx + 4],
+                )
+                for idx, element in enumerate(elements)
+                if element.title == "Channel"
+            ]
+        except IndexError as e:
+            msg = "Unable to get channel elements from Measurement section."
+            raise AllotropeConversionError(msg) from e
+
+    def try_element_or_none(self, title: str) -> Optional[MeasurementElement]:
+        for element in self.elements:
+            if element.title == title:
+                return element
+        return None
+
+    def get_number_of_averages(self) -> Optional[float]:
+        number_of_flashes = self.try_element_or_none(self.number_of_flashes)
+        if number_of_flashes is None:
+            return None
+        return try_float(number_of_flashes.value, self.number_of_flashes)
+
+    def get_detector_distance(self) -> Optional[float]:
+        detector_distance = self.try_element_or_none(self.detector_distance)
+        if detector_distance is None:
+            return None
+        return try_float(detector_distance.value, self.detector_distance)
+
+    def get_scan_position(self) -> Optional[ScanPositionSettingPlateReader]:
+        position = self.try_element_or_none(self.position)
+        if position is None:
+            return None
+
+        return assert_not_none(
+            SCAN_POSITION_CONVERTION.get(position.value),
+            msg=f"'{position.value}' is not a valid scan position, expected TOP or BOTTOM.",
+        )
+
+    def get_emission_wavelength(self) -> Optional[float]:
+        emission_wavelength = self.try_element_or_none(self.emission_wavelength)
+        if emission_wavelength is None:
+            return None
+        return try_float(emission_wavelength.value, self.emission_wavelength)
+
+    def get_excitation_wavelength(self) -> Optional[float]:
+        excitation_wavelength = self.try_element_or_none(self.excitation_wavelength)
+        if excitation_wavelength is None:
+            return None
+        return try_float(
+            excitation_wavelength.value.removesuffix("nm"), self.excitation_wavelength
+        )
+
+    def get_focus_height(self) -> Optional[float]:
+        focus_height = self.try_element_or_none(self.focus_height)
+        if focus_height is None:
+            return None
+        return try_float(focus_height.value, self.focus_height)
+
+
+@dataclass(frozen=True)
+class Data:
+    version: str
+    background_info: BackgroundInfo
+    results: Results
+    analysis_results: list[AnalysisResult]
+    measurement_info: MeasurementInfo
+    plate_type: PlateType
+    platemap: Platemap
+    measurements: Measurements
+
+    def get_experiment_type(self) -> str:
```

Comment:
Generally, you should not need/want to create accessor classes for nested dataclasses, you can just access them directly. 

e.g. replace `data.get_experiment_type()` with `data.background_info.experiment_type`

The dataclass names/structure should be self documenting and simple access, removing the need to these functions.